### PR TITLE
Demo UI

### PR DESCRIPTION
**Pull request by matthewearl**
Add a user interface for seeking through demos:

image

Implementation notes:
- The UI (defined in `cl_demoui.c`) primarily drives new `demoseek` and `demoskip` commands which skip within and between levels respectively. 
- I added a new parser `demoseekparse.c` to pull out the file offsets at the start of each map, which `demoskip` uses to seek the open demo file to.  
- Seeking forwards within levels is done by fetching messages until `cl.mtime[0]` reaches the target seek time.
- Seeking backwards within levels is done in a similar way, but using the rewind mechanism.  I've replaced references to `cl_demorewind.value` with a generic `CL_DemoRewind()` function to catch both the seeking case and the reverse demo case.
- There are a few glitches that can occur when seeking around, which I've attempted to mitigate in `EndSeek`.  There is still a problem with looping sounds however --- rewind when a lift sound is in progress (for example), and the lift sound will never stop.  This is generally a problem for destructive operations, which we might want to fix in future with a stack like system as with Sphere's intermission screen fix.
- `demoseekparse.c` also pulls out finish times, which is used to keep the ghost splits in sync when seeking between levels. The finish time is also used to add a small indicator onto the seek bar.

### CODE REVIEW

File: trunk/cl_demo.c [hunk 1]
```diff
@@ -21,26 +21,22 @@ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
...
-// .dz playback
-#ifdef _WIN32
-static	HANDLE	hDZipProcess = NULL;
-#else
-#include <errno.h>
-//#include <sys/types.h>
-#include <sys/wait.h>
-#include <unistd.h>
-static qboolean hDZipProcess = false;
-#endif
```
Comment (matthewearl) [inline trunk/cl_demo.c:42]: Unrelated to the UI, but all these vars were unused, so I got rid of them.

**Comment by j0zzz** [comment]
This is amazing, well done!

I definately want to write about this in the project page.
I'm planning to create just a blank page and then you can write about the feature (if you don't mind) just as you did with the ghost feature.

**Comment by j0zzz** [comment]
Updated the github project page with the new Demo UI feature.
Create a new subfeature for the documentation here: `docs/_includes/subfeatures/demo-ui.md`

**Comment by matthewearl** [comment]
Awesome, I'll fill in the placeholders soon

**Comment by matthewearl** [comment]
I've made a PR with the updated docs here: https://github.com/j0zzz/JoeQuake/pull/143

**Comment by j0zzz** [comment]
Thanks again Matt. I'll release a new version soon, so that everyone can try out the new player UI. Absolutely loving it. JoeQuake got a youtube player :)

**Pull request merged**